camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlMultiOutputCommitter.java

```diff
@@ -149,10 +146,17 @@ public void commitTask(TaskAttemptContext context) throws IOException {
   }
 
   protected void commitFile(JobContext job, Path source, Path target) throws IOException {
-    log.info(String.format("Moving %s to %s", source, target));
-    if (!FileSystem.get(job.getConfiguration()).rename(source, target)) {
-      log.error(String.format("Failed to move from %s to %s", source, target));
-      throw new IOException(String.format("Failed to move from %s to %s", source, target));
+    FileSystem fs = FileSystem.get(job.getConfiguration());
+
+    if (!fs.exists(target)) {
```

Comment:
I think it would be better to rename and then catch the exception as opposed to doing an existence check because rename will be atomic where as this won't.